Makefile

```diff
@@ -1,9 +1,359 @@
 # #############################################################################
-# Simple Makefile to build and test Lightweight Logging Library (l3) sources
+# L3: Simple Makefile to build and test Lightweight Logging Library sources
```

Comment:
Massively reworked Makefile.

(We've used this previously at VMware, SplinterDB, and this scheme has worked very well.)
I have cloned this version from LOC's Makefile, which itself was derived from SplinterDB's Makefile.